routes/errortracker.js

```diff
@@ -0,0 +1,268 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *objects used
+ *Objects are identical to the objects used in previous errortracker.go
+ * @typedef errorRequestMeta {
+	HTTPReferrer: string,
+	HTTPUserAgent: string
+    }
+  @typedef errorRequest {
+	URL:  string,
+	Method: string,
+	Meta: errorRequestMeta
+  }
+  @typedef errorEvent {
+	Application: string,
+	AppID; string,
+	Environment: string,
+	Version: string,
+	Message: string,
+	Exception: string,
+	Request: *errorRequest,
+	Filename: string,
+	Line: int32,
+	Classname: string,
+	Function: string,
+	Severity: string
+   }
+ * Handle error requests from clients and log them.
+ */
+// AppEngine project ID
+const express = require('express');
+const router = express.Router();
+const logging = require('@google-cloud/logging');
+const winston = require('winston');
+const statusCodes = require('http-status-codes');
+const Math = require('../tests/Math');
+const url = require('url');
+const projectId ='';
+const logName = 'javascript.errors';
+/**
+ * ERROR_LEVELS
+ * @enum {string}
+ */
+const ERROR_LEVELS = {
+    DEFAULT: 'Default',
+    DEBUG: 'Debug',
+    INFO: 'Info',
+    NOTICE: 'Notice',
+    WARNING: 'Warning',
+    ERROR: 'Error',
+    CRITICAL: 'Critical',
+    ALERT: 'Alert',
+    EMERGENCY: 'Emergency',
+};
+
+/**
+ * Filter exceptions in an array to prevent them from being logged
+ * @param errorEvent
+ * @returns {boolean}
+ */
+function isFilteredMessageOrException(errorEvent) {
+    let filteredMessageOrException = ['stop_youtube',
+        'null%20is%20not%20an%20object%20(evaluating%20%27elt.parentNode%27)'];
+    return filteredMessageOrException.some(function filter(msg) {
+        return errorEvent.message.includes(msg) || errorEvent.exception.includes(msg);
+    });
```

Comment:
Check out the Google JS style guide for formatting guidance: https://google.github.io/styleguide/jsguide.html#formatting

Use existing AMP code as a reference. Try to make it look as close to the AMP style code as possible (consistent indentation, etc.). Let's talk more about this in person.